eo-maven-plugin/src/main/java/org/eolang/maven/SafeMojo.java

```diff
@@ -285,4 +299,33 @@ private static void closeTojos(final Closeable res) throws MojoFailureException
         }
     }
 
+    /**
+     * Interrupts thread if time limit is reached.
+     *
+     * @param thread Thread to interrupt.
+     * @param sec Time limit in seconds.
+     */
+    private static void waitAndInterrupt(final Thread thread, final int sec) {
+        Executors.newSingleThreadExecutor().submit(
+            () -> {
+                try {
+                    Thread.sleep(TimeUnit.SECONDS.toMillis(sec));
+                    synchronized (thread) {
+                        if (thread.isAlive()) {
+                            thread.interrupt();
+                        }
+                    }
+                } catch (final InterruptedException ex) {
```

Comment:
@volodya-lombrozo I'm sure you know this, but still, please read: https://www.yegor256.com/2015/10/20/interrupted-exception.html